### DISCUSSION TITLE
Using debug and release side by side on windows

### DISCUSSION THREAD

**Discussion by Peaj** [dotnet/maui#18630] [Q&A] [answered]
I am developing an app that I am also trying to use myself on the same windows pc I am developing on.
However every time I do a debug build Visual Studio warns me it will override the existing already installed version of the application:

"The app "xxx.xxx.xxx" is already installed on this machine.
If you continue with the current deployment, the existing app
will be uninstalled and the app's current state will be deleted,
Are you sure you want to continue?"

Agreeing to this will overwrite the published app running on my PC which makes it unusable.

So far the only solution I see is changing the ApplicationId for debug builds.
I tried to change the ApplicationId in the csproj file based on the build configuration, but I figured it is completely ignored by windows builds.
The only thing that works is to change the Identity.Name in the Platforms/Windows/Package.appxmanifest.
I dont understand why the app id (and also the app version) has to be configured and updated in multiple different files.
But the bigger problem is that I cannot find a way to set this programmatically by build configuration as this file does not allow conditional property groups.

My current workaround is to change this identifier manually and be really cautious not to accidentally commit it which is a really bad workflow.

I really wonder if I am the only one having a problem with that.
Is there any workflow or setting that can make this work?

**Comment by baaaaif** [comment, marked as answer]
We are using something like the following Condition to work against this problem

```
<ItemGroup Condition="'$(TargetFramework)'=='$(NetVersion)-windows10.0.19041.0'">
<AppxManifest Include="..\..\AppLabels\$(_AppLabelFolderName)\Package.appxmanifest" Link="Platforms\Windows\Package.appxmanifest" />
</ItemGroup>
```
_AppLabelFolderName is defined by different release configurations
NetVersion is net7 or net8

**Reply by Peaj** [reply]
That seems like the best solution so far. But wouldnt that mean that I have to have a copy of the package.appxmanifest for each configuration? That would be even more annoying when changing sth like version number because it has to be updated across min3 different files. I guess it is not possible to override only part of an appxmanifest is it?

**Reply by baaaaif** [reply]
Yes, you'll need copies of your appxmanifest
If your package.appxmanifest version number is 0.0.0.0 it will be created from ApplicationDisplayVersion and Version

**Reply by Peaj** [reply]
Wow thanks. That actually solves my second biggest problem with the MAUI config. It was always annoying to change that version in multiple places.

**Reply by Peaj** [reply]
I just figured out that this is also true for the package name. It should not be changed at all in the appxmanifest. If it is set to the default "maui-package-name-placeholder" or empty it will be replaced with the identifier from the csproj config.
This also enables usage of conditional configuration in the csproj file.

**Reply by kdotdk** [reply]
> If it is set to the default "maui-package-name-placeholder" or empty it will be replaced with the identifier from the csproj config.
Which property in the csproj do you change update change identity name of the manifest?
